File: JAVA/Remove Duplicates.java
class Solution {
    String removeDups(String str) {
        Set<Character>  charSet = new  LinkedHashSet<>();
        char[]  strArray = str.toCharArray();
        for(char eachLetter : strArray){
            charSet.add(eachLetter);
        }
        StringBuilder  sb = new StringBuilder();
        for(char ch : charSet){
            sb.append(ch);
        }
        return sb.toString();
    }
}

File: C++/Remove Duplicates.cpp
class Solution {
  public:
    string removeDups(string str) {
        int n = str.size(), len = 0, mask = 0;
        for(int i = 0; i < n; i++) {
            int ch = str[i] - 'a';
            if((mask >> ch) & 1) continue;
            mask |= (1 << ch);
            str[len++] = ch + 'a';
        }
        return str.substr(0, len);
    }
};
